tests/sqlparser_snowflake.rs

```diff
@@ -1010,15 +1013,9 @@ fn test_snowflake_stage_object_names() {
     ];
     let mut allowed_object_names = vec![
         ObjectName(vec![Ident::new("my_company"), Ident::new("emp_basic")]),
-        ObjectName(vec![Ident::new("@namespace"), Ident::new("%table_name")]),
-        ObjectName(vec![
-            Ident::new("@namespace"),
-            Ident::new("%table_name/path"),
-        ]),
-        ObjectName(vec![
-            Ident::new("@namespace"),
-            Ident::new("stage_name/path"),
-        ]),
+        ObjectName(vec![Ident::new("@namespace.%table_name")]),
```

Comment:
why were these tests changed?

Reply:
Tbh, I don't know how much big difference that makes, I don't even know if `.` in Snowflake makes a difference or not when checking the test cases. I was even thinking about keeping the test assertions as they were and just ignoring the tests until that gets fixed properly.

Reply:
I think that if `.` is a separator (assuming ObjectName was used here), then possibly we could keep `At`,`Mod` etc tokens, and somehow concat them to get SF identifiers, I don't know Snowlake that much to make a decision.

Reply:
I am wary of changing tests when we don't understand the reason for why they changed

Reply:
Updated PR to keep tokenising as before and instead add special handling in stage name parsing.